lib/AST/DeclContext.cpp

```diff
@@ -300,6 +300,17 @@ PackageUnit *DeclContext::getPackageContext(bool lookupIfNotCurrent) const {
   return nullptr;
 }
 
+bool DeclContext::bypassResilienceInPackage(bool isForPackageDecl) const {
+  // Bypassing resilience checks only applies to package types (and possibly
+  // public types in a package in the future). Allowed only if opted-in for
+  // bypassing checks, client and defining module are in the same package,
+  // and defining module is a binary module.
+  return isForPackageDecl &&
+         getASTContext().LangOpts.EnableBypassResilienceInPackage &&
```

Comment:
I think this will not always be set correctly -- we should pass in the ModuleDecl doing the access instead. Then you can also avoid converting the Identifier to a std::string, which is inefficient -- this check is going to be executed a lot